### ISSUE TITLE
Lazer failing to enter Exclusive Fullscreen mode

### ISSUE THREAD

**Issue by Zyro1331** [ppy/osu#24659]
### Type

Game behaviour

### Bug description

Lazer keeps failing to enter Exclusive Fullscreen on Windows 11. 
I use a dual 1080p monitor setup with my Secondary monitor above my Primary display. _The only other display connected to my GPU is a VR Headset which isn't in use, nor recognized as a valid display by Windows._ 

Switching between Fullscreen and Windowed modes in settings appears to do nothing, including using Alt + Enter to change modes. 
This issue occurs with both OpenGL and Direct3D 11 rendering APIs. 

Here is my Graphics Card details;
```
AMD Radeon RX 6750 XT
Driver version:	31.0.21023.2010
Driver date:	7/20/2023

Exact driver version (According to AMD):
Driver Version 23.10.23.02-230720a-394441C-AMD-Software-Adrenalin-Edition
```
Windows details;
```
Edition: Windows 11 Pro
Version: 22H2
OS build: 22621.2215
```

A video has been attached showcasing the exact issue in detail, along with the runtime log files. 

### Screenshots or videos

https://github.com/ppy/osu/assets/49619936/ab2cafff-eb00-43a0-8e3f-9f1336a9871d

My apologies for the quality, just sadly not much I can do to downsize a 4k video to 10mb without making text incredibly difficult to see.

### Version

2023.815.0

### Logs

[network.log](https://github.com/ppy/osu/files/12443502/network.log)
[performance.log](https://github.com/ppy/osu/files/12443503/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12443504/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12443505/updater.log)

**Comment by Zyro1331** [comment]
**Update on this issue**
It appears to be driver dependent. I noticed that inside Windows settings that my monitor refresh-rate and resolution was basically stuck in place, unable to be changed. This aligns with the missing display mode alerts in the [runtime.log](https://github.com/ppy/osu/files/12443504/runtime.log) file:
```
2023-08-25 21:01:33 [verbose]: Updated display mode to desktop resolution: 1920x1080@120, SDL_PIXELFORMAT_RGB888
2023-08-25 21:01:33 [verbose]: Display at index (0) (1920x1080) has no display modes. Fullscreen might not work.
2023-08-25 21:01:33 [verbose]: Display at index (1) (1920x1080) has no display modes. Fullscreen might not work.
```
I recently wiped my AMD Drivers using DDU and it appears to have fixed the Exclusive Fullscreen issue. 
Likely later on in development, a windows-specific fallback for detecting the display and forcing the Windows API to Fullscreen at current resolution should be added later since Stable seemed to link up just fine even while this problem was occurring. 

Thanks for your patience, and reading into this issue.
